dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala

```diff
@@ -370,97 +373,253 @@ case class BinaryOutcomeDLCWithSelf(
+        refundTx <- refundTxF
+        _ <- fundingTxPublishedF
+      } yield {
+        SetupDLC(fundingTx,
+                 fundingSpendingInfo,
+                 cetWinLocal,
+                 cetLoseLocal,
+                 cetWinRemote,
+                 cetLoseRemote,
+                 refundTx)
+      }
+    }
+  }
 
-        val cetReadyForPublish =
-          fundingTxPublishedF.flatMap(_ => cetAndPrivKeyF)
-
-        cetReadyForPublish.flatMap {
-          case (cet, cetPrivKey) =>
-            val cetPublishedF = messengerOpt match {
-              case Some(messenger) =>
-                messenger
-                  .sendTransaction(cet)
-                  .flatMap(_ => messenger.waitForConfirmations(blocks = 6))
-              case None => FutureUtil.unit
-            }
+  def publishClosingTx(
+      cetPublishedF: Future[Unit],
+      cetOutput: TransactionOutput,
+      privKey: ECPrivateKey,
+      spendingInfo: BitcoinUTXOSpendingInfo,
+      isLocal: Boolean,
+      messengerOpt: Option[BitcoinP2PMessenger]): Future[Transaction] = {
+    // Construct Closing Transaction
+    val txBuilder = BitcoinTxBuilder(
+      Vector(
+        TransactionOutput(cetOutput.value,
+                          P2PKHScriptPubKey(privKey.publicKey))),
+      Vector(spendingInfo),
+      feeRate,
+      changeSPK,
+      network
+    )
 
-            val output = cet.outputs.head
+    val spendingTxF = txBuilder.flatMap(subtractFeeAndSign)
+
+    spendingTxF.foreach(
+      tx =>
+        logger.info(
+          s"${if (isLocal) "Local" else "Remote"} Closing Tx: ${tx.hex}"))
+
+    val spendingTxPublishedF = spendingTxF.flatMap { spendingTx =>
+      cetPublishedF.flatMap { _ =>
+        messengerOpt match {
+          case Some(messenger) =>
+            messenger
+              .sendTransaction(spendingTx)
+              .flatMap(_ => messenger.waitForConfirmations(blocks = 6))
+          case None => FutureUtil.unit
+        }
+      }
+    }
 
-            // Spend the true case on the correct CET
-            val cetSpendingInfo = ConditionalSpendingInfo(
-              TransactionOutPoint(cet.txIdBE, UInt32.zero),
-              output.value,
-              output.scriptPubKey.asInstanceOf[ConditionalScriptPubKey],
-              Vector(cetPrivKey, ECPrivateKey(oracleSig.s)),
-              HashType.sigHashAll,
-              ConditionalPath.nonNestedTrue
-            )
+    spendingTxF.flatMap { spendingTx =>
+      spendingTxPublishedF.map { _ =>
+        spendingTx
+      }
+    }
+  }
 
-            val finalPrivKey = if (local) {
-              finalLocalPrivKey
+  /** Constructs and executes on the unilateral spending branch of a DLC
+    *
+    * @return Each transaction published and its spending info
+    */
+  def executeUnilateralDLC(
+      oracleSigF: Future[SchnorrDigitalSignature],
+      local: Boolean,
+      messengerOpt: Option[BitcoinP2PMessenger] = None): Future[DLCOutcome] = {
+    setupDLC(messengerOpt).flatMap {
```

Comment:
setup shouldn't be happening in the `executeUnilateralDLC()` method. You are co-mingling test code and source code.